misk/web/@misk/components/src/components/TextHTMLOrElementComponent.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react"
+
+/**
+ * <TextHTMLOrElementComponent content={<div>item</div>} length={35} />
+ * <TextHTMLOrElementComponent content={"Test"} length={35} />
+ * <TextHTMLOrElementComponent content={'<a href="http://squareup.com/">Test</a>'} length={35} />
+ *
+ * Renders passed in text, HTML-parseable string, or React element with optional length constraints for text.
+ */
+
+export interface ITextHTMLOrElementProps {
+  content: string | Element | JSX.Element
+}
+
+export const TextHTMLOrElementComponent = (
+  props: ITextHTMLOrElementProps
+): JSX.Element => {
+  const { content }: any = props
+  let FormattedContent: any = content
+
+  if (typeof content === "string" && !content.startsWith("<")) {
+    FormattedContent = <span>{content}</span>
+  } else if (typeof content === "string" && content.startsWith("<")) {
```

Comment:
When would we want to pass in an html string instead of a `JSX.Element`?

Reply:
An unlikely use case but would allow for example server side rendering of navbar items to come from Misk server, static definitions, and other places that aren't in the same Typescript project.